Added default text colour, background mode members to the HTML parser…

vgromov opened the pull request:
… class, to allow non hard-coded colours to be used when generating parsed HTML cell objects.

Added SetHtmlText method to wxHtmlDCRenderer class, taking const wxHtmlRenderingState& initialState parameter utilized to change base HTML Parser cell colours.

vgromov committed:
Added default text colour, background mode members to the HTML parser class, to allow non hard-coded colours to be used when generating parsed HTML cell objects.

Added SetHtmlText method to wxHtmlDCRenderer class, taking const wxHtmlRenderingState& initialState parameter utilized to change base HTML Parser cell colours.

vadz commented:
Sorry, but I'm not sure at all why is this useful, how is using the new function better than just calling `SetLinkColor()` etc before calling `SetHtmlText()`?

In fact, I think that setting colours is completely orthogonal to the purpose of `SetHtmlText()`, so I don't think we want to apply this in its current state.

vgromov commented:
> Sorry, but I'm not sure at all why is this useful, how is using the new function better than just calling `SetLinkColor()` etc before calling `SetHtmlText()`?
Sorry, I was not quite clear on the case. I use wxHtmlDCRenderer directly in a few custom painted controls, which may use small subset of html markup to format misc. units of measurement, containing super/sub-script symbols, etc. _except for_ explicit colour markup. I want these HTML fragments to blend seamlessly into the rest of control's text colouring style.
HTML renderer just have _no way_ to control the colouring of HTML text. It just receives default hard-coded colour values set upon internal parser initialization, called from within SetHtmlText. So, despite the way other non-html text may look in control, the html portion, unless explicitly utilize colour markup, would receive black text on transparent background.

When we look at wxHtmlWinParser interface, it seemingly allows to set a bunch of colours using `    const wxColour& GetLinkColor() const { return m_LinkColor; }
    void SetLinkColor(const wxColour& clr) { m_LinkColor = clr; }
    const wxColour& GetActualColor() const { return m_ActualColor; }
    void SetActualColor(const wxColour& clr) { m_ActualColor = clr ;}
    const wxColour& GetActualBackgroundColor() const { return m_ActualBackgroundColor; }
    void SetActualBackgroundColor(const wxColour& clr) { m_ActualBackgroundColor = clr;}
    int GetActualBackgroundMode() const { return m_ActualBackgroundMode; }
    void SetActualBackgroundMode(int mode) { m_ActualBackgroundMode = mode;}`
but, looking closely at actual Parse(string) code, we will see that anything set by these methods prior to Parse call, would be _reset_ to hard-coded initial values in InitParser. So there is no way to specify _custom_ starting base text attributes there as well.
After Parse is complete, we already cannot control parsed cells drawing, unless these are in special IN_SELECTION rendering state. All colours used by rendering routine for cells with normal state, use colour values assigned to cells during parsing process, so no luck there either.
 What I did, was 

1. added a few interface methods allowing custom base text attributes assignment to Parser, which, if set, will be used in InitParser instead of hard-coded values. 
2. created an additional SetHtmlText call form, allowing to pass _non hard-coded_ HTML text attributes to the underlying parser, otherwise inaccessible to the programmer through wxHtmlDCRenderer interface. An additional text attributes parameter is quite obviously called initialState, to explain, that these parameters are used upon Html Parser initialization stage.

> In fact, I think that setting colours is completely orthogonal to the purpose of `SetHtmlText()`, so I don't think we want to apply this in its current state.

Hope I provided a detailed explanation of the problem and implemented solution. 
If you would be so kind to offer better way to solve this, without changing current implementation, or using explicit colour tags in markup, I'm eager to know.

vadz commented:
> looking closely at actual Parse(string) code, we will see that anything set by these methods prior to Parse call, would be reset to hard-coded initial values in InitParser. 

I think this is the real problem, we shouldn't be setting all this stuff in `InitParser()`. Why don't we do it in the ctor instead? Maybe we need some (private) `InitAppearance()` that would be called from both ctor and `DoneParsing()`, but I'm not even sure about this: it probably would make sense for the values of these variables to be used for the next call to `Parse()` too, at least I don't see why not.

So could you please experiment with just moving `m_LinkColor` etc initialization to the ctor?

@vslavik It's probably too much to hope you remember anything about this 20 year old code, but please let me know if I'm missing something here if you do. TIA!
